go-controller/pkg/ovn/master.go

```diff
@@ -1030,6 +1030,14 @@ func (oc *Controller) addNode(node *kapi.Node) ([]*net.IPNet, error) {
 		return nil, err
 	}
 
+	chassisID, err := util.ParseNodeChassisIDAnnotation(node)
+	if err == nil {
+		// delete stale chassis in SBDB if system-id of the specific node has changed.
+		if err = oc.deleteNodeChassis(node.Name, chassisID); err != nil {
```

Comment:
@cathy-zhou  I am thinking if we should do this instead?

> err, match := oc.checkNodeChassisMatch(node.Name, chassisID)
> If err == nil && !match {
>      err = oc.deleteNodeChassis(node.Name); err != nil {
>          // Record a event here so that we will know that there is an isssue
>          // with this node?
>      }
> }

My thinking is that this is a serious issue and just logging will not get the attention from the deployer. With this error, ovn-controller stalls and doesn't proceed.

Reply:
Thanks. I've made the change. Please take another look.